WebDriver: Get Element Rect should not round to integer values

patrickangle opened the pull request:
#### ac88ff2a875c5707cb4219de247d20a0192d576a
<pre>
WebDriver: Get Element Rect should not round to integer values
<a href="https://bugs.webkit.org/show_bug.cgi?id=248556">https://bugs.webkit.org/show_bug.cgi?id=248556</a>
rdar://100588620

Reviewed by BJ Burg.

The x/y/width/height of the element rect is specified to be in &quot;CSS Pixels&quot;, which is a decimal number of pixels. We
should as such not round to the nearest containing IntRect.

This progresses the following WPT tests:
webdriver/tests/get_element_rect/get.py::test_basic
webdriver/tests/get_element_rect/user_prompts.py::test_accept[capabilities0-alert-None]
webdriver/tests/get_element_rect/user_prompts.py::test_accept[capabilities0-confirm-True]
webdriver/tests/get_element_rect/user_prompts.py::test_accept[capabilities0-prompt-]
webdriver/tests/get_element_rect/user_prompts.py::test_dismiss[capabilities0-alert-None]
webdriver/tests/get_element_rect/user_prompts.py::test_dismiss[capabilities0-confirm-False]
webdriver/tests/get_element_rect/user_prompts.py::test_dismiss[capabilities0-prompt-None]

* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::computeElementLayout):
(WebKit::WebAutomationSession::viewportInViewCenterPointOfElement):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::computeElementLayout):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.messages.in:

Canonical link: <a href="https://commits.webkit.org/257498@main">https://commits.webkit.org/257498@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e433086c6fa76cb6ef976335d3adb9a5a0d31028

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98979 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8186 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32138 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108386 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8737 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85599 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91504 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106354 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104691 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6618 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90202 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33642 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/88440 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21547 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76495 "Found 4 new API test failures: /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/network-error, /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/console-api, /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/js-exception, /WebKitGTK/TestConsoleMessage:/webkit/WebKitConsoleMessage/security-error (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2087 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23063 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1992 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45499 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5156 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6954 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42537 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3405 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

burg reviewed (approved):
LGTM, nice fix! (I believe this was originally added for Selenium compat?)

patrickangle committed:
WebDriver: Get Element Rect should not round to integer values
https://bugs.webkit.org/show_bug.cgi?id=248556
rdar://100588620

Reviewed by BJ Burg.

The x/y/width/height of the element rect is specified to be in "CSS Pixels", which is a decimal number of pixels. We
should as such not round to the nearest containing IntRect.

This progresses the following WPT tests:
webdriver/tests/get_element_rect/get.py::test_basic
webdriver/tests/get_element_rect/user_prompts.py::test_accept[capabilities0-alert-None]
webdriver/tests/get_element_rect/user_prompts.py::test_accept[capabilities0-confirm-True]
webdriver/tests/get_element_rect/user_prompts.py::test_accept[capabilities0-prompt-]
webdriver/tests/get_element_rect/user_prompts.py::test_dismiss[capabilities0-alert-None]
webdriver/tests/get_element_rect/user_prompts.py::test_dismiss[capabilities0-confirm-False]
webdriver/tests/get_element_rect/user_prompts.py::test_dismiss[capabilities0-prompt-None]

* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::computeElementLayout):
(WebKit::WebAutomationSession::viewportInViewCenterPointOfElement):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::computeElementLayout):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.messages.in:

Canonical link: https://commits.webkit.org/257498@main
